File: src/main/java/preprocessing/PreprocessorImp.java
package preprocessing;

import org.apache.commons.lang3.StringUtils;

import java.util.Objects;

public class PreprocessorImp implements Preprocessor {

    public String getPreprocessedText(final String rawText) {
        Objects.requireNonNull(rawText, "Raw text cannot be null");
        String preprocessedText = rawText.replaceAll("\\b(?i)EC\\b", " ");
        return StringUtils.normalizeSpace(preprocessedText);
    }
}

File: src/main/java/preprocessing/PreprocessorImp.kt
package preprocessing

class PreprocessorImp : Preprocessor {

    override fun getPreprocessedText(rawText: String): String {
        val preprocessedText = rawText.replace("\\b(?i)EC\\b".toRegex(), " ")
        return preprocessedText.replace("\\s+".toRegex(), " ").trim()
    }
}
